cron/postfix_logs.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+#TO DO 
+#	- Delete old logs
+
+#check if maillog directory exists
+if [ ! -d ../maillog ]
+then
+        #create direcotry
+	mkdir ../maillog
+fi
+
+
+data=`date --date='1 day ago' +%F`
+
+mysql -u root --skip-column-names --execute="USE postfix; SELECT username,domain FROM domain_admins WHERE active=1 AND domain <> 'ALL';" | while read username domain
+	    do
+
+		#check if folder for domain doesn t exists
+		if [ ! -d ../maillog/$domain ]
+		then
+    			echo "Directory does not exists for "$domain
+			#create direcotry
+			mkdir ../maillog/$domain			
+		fi
+
```

Comment:
with your use of mail.log.1 you presumably need to run this script fairly soon after logrotate has run?

Reply:
Yes. The script will be run fairly soon after logrotate.

The admins can not have access to logs from the current day.  For the current day, the script should be executed maybe every one hour, and mail.log.1 changed to mail.log.

Reply:
OK. Perhaps I should look at rsyslog and how it puts stuff into a database....

Reply:
I tried to move logs in the database, but in some cases will be a large table. For example(in some cases),  2.000.000 logs/24h(for each domain), and I don't know what will be the best method to allow an admin to search for a specific log.

Reply:
> OK. Perhaps I should look at rsyslog and how it puts stuff into a database....

I don t know how to do that

Reply:
I'd assume an admin would want to search for log messages to/from a specific address?

Reply:
yep. e-mail address and  maybe ip, datetime

Reply:
With regards to the volume of logs -> for a small server (like mine) - I suspect a MySQL/PostgreSQL database would cope with the volume fine. For someone with more logs, I'd hope they'd use something better like ElasticSearch.

I've just had a look at the 'rsyslog-pgsql' package (Debian).

The main problem with rsyslog-XXsql will be that the log entries in the systemevents table is just one row per line of the log file, with no extra structure ... so you'd (probably?) end up wanting to parse / process these records in order to get them more structured (doing a "select * From systemevents where syslogtag like 'postfix%' and message like '%from=<foo@bar.com%';" to find incoming mail from foo@bar.com isn't going to work well.

It would work fine if you know what time frame you're looking for and you've got a fairly distinct pattern (e.g "what postfix messages were logged involving icloud.com between 6:30 and 7pm last night?") .... but once you identify a matching line/message, you'd want to parse that to retrieve the postfix identifier, and then re-query the database to fetch the matching lines - e.g.

```
Syslog=# \x
Expanded display is on.
Syslog=# select * From systemevents where syslogtag like 'postfix%' and message like '%from=<david@exam%';
-[ RECORD 1 ]------+-----------------------------------------------------------------------------
id                 | 28
customerid         | 
receivedat         | 2022-03-21 21:54:13
devicereportedtime | 2022-03-21 21:54:13
facility           | 2
priority           | 6
fromhost           | mail
message            |  6DF2A86E5: from=<david@example.co.uk>, size=939, nrcpt=1 (queue active)
ntseverity         | 
importance         | 
eventsource        | 
eventuser          | 
eventcategory      | 
eventid            | 
eventbinarydata    | 
maxavailable       | 
currusage          | 
minusage           | 
maxusage           | 
infounitid         | 1
syslogtag          | postfix/qmgr[754655]:
eventlogtype       | 
genericfilename    | 
systemid           | 
-[ RECORD 2 ]------+-----------------------------------------------------------------------------
id                 | 35
customerid         | 
receivedat         | 2022-03-21 21:54:14
devicereportedtime | 2022-03-21 21:54:14
facility           | 2
priority           | 6
fromhost           | mail
message            |  50A32870C: from=<david@example.co.uk>, size=1692, nrcpt=2 (queue active)
ntseverity         | 
importance         | 
eventsource        | 
eventuser          | 
eventcategory      | 
eventid            | 
eventbinarydata    | 
maxavailable       | 
currusage          | 
minusage           | 
maxusage           | 
infounitid         | 1
syslogtag          | postfix/qmgr[754655]:
eventlogtype       | 
genericfilename    | 
systemid           | 

```


and then e.g. 

`select * From systemevents where syslogtag like 'postfix%' and message like ' 50A32870C%';`

Reply:
I think with ElasticSearch you will need more resources. For example, if you have 1-2.000.000 logs/24h and you have 40-50 domains and you allow the postmasters to search logs in the last 30 days.  I think you will need to have a lot of RAM .

In my opinion, if we will create a folder for each domain where we will put the logs will be more simple and you will not have to give more resources.  Also, it will be more simple to configure.

Reply:
Maybe a log listener that copies in 'real time' logs from ../var/log/mail.log in ../postfixadmin/maillog/domain/ will be a more better?

Reply:
Maybe a log listener that copies in 'real time' logs from ../var/log/mail.log in ../postfixadmin/maillog/domain/ will be more better?

Reply:
or every 1 hour/ 30 min

Reply:
e.g. https://superuser.com/questions/181517/how-to-execute-a-command-whenever-a-file-changes ?

Reply:
> e.g. https://superuser.com/questions/181517/how-to-execute-a-command-whenever-a-file-changes ?

yes

Reply:
https://github.com/bnchdan/postfixadmin_logs/blob/main/main.sh maybe? it is not yet finished

Reply:
something like https://github.com/bnchdan/postfixadmin_logs/blob/main/main.sh  ?